### PR TITLE
[SDX-1041] Create box component

### PR DESCRIPTION
**Pull request by ahollesen**
- moved BaseDropdown to general base package
- added BaseBox and Box
- added BodyMaxWidthLayout
- using BaseBox inside BodyPanel

### CODE REVIEW

File: src/framework/box/Box.jsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
+import classNames from 'classnames';
```
Comment (mohsinhijazee) [inline src/framework/box/Box.jsx]: Probably import from `classnames/dedupe` to avoid class name duplicates.

File: src/framework/box/Box.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        const testCase = TestCaseFactory.create(Box, props);
+        expect(testCase.dom.className).toContain(props.classes);
+      });
+
+      it('don\'t replace "box" class', () => {
+        const props = {
+          classes: 'testClass',
+        };
+        const testCase = TestCaseFactory.create(Box, props);
+        expect(testCase.dom.className).toContain('box');
```
Comment (mohsinhijazee) [inline src/framework/box/Box.spec.jsx]: Probably add a test for duplicate CSS classes if makes sense. Optional.

File: src/framework/box/Box.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    describe('classes', () => {
+      it('are added to the element', () => {
+        const props = {
+          classes: 'testClass',
+        };
+        const testCase = TestCaseFactory.create(Box, props);
+        expect(testCase.dom.className).toContain(props.classes);
+      });
+
+      it('don\'t replace "box" class', () => {
```
Comment (mohsinhijazee) [inline src/framework/box/Box.spec.jsx]: Replace `don't` with `doesn't`

File: src/framework/base/box/BaseBox.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+
+import classNames from 'classnames/dedupe';
```
Comment (jenslonkowski) [inline src/framework/base/box/BaseBox.jsx]: Why do we need dedupe here?

File: src/framework/base/box/BaseBox.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        };
+        const testCase = TestCaseFactory.create(BaseBox, props);
+        expect(testCase.dom.getAttribute('data-id')).toBe(props.dataId);
+      });
+    });
+
+    describe('roundedCorners', () => {
+      it('when false doesn\'t add \'roundedCorners\' class', () => {
+        const props = {
+          roundedCorners: false,
```
Comment (jenslonkowski) [inline src/framework/base/box/BaseBox.spec.jsx]: The required prop `classes` is missing.

File: src/framework/base/box/BaseBox.spec.jsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+          classes: '',
+          dataId: 'dataId',
+        };
+        const testCase = TestCaseFactory.create(BaseBox, props);
+        expect(testCase.dom.getAttribute('data-id')).toBe(props.dataId);
+      });
+    });
+
+    describe('roundedCorners', () => {
+      it('when false doesn\'t add \'roundedCorners\' class', () => {
```
Comment (jenslonkowski) [inline src/framework/base/box/BaseBox.spec.jsx]: `roundedCorners` can be in double quotation marks.

File: src/framework/base/box/BaseBox.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        const props = {
+          roundedCorners: false,
+        };
+        const testCase = TestCaseFactory.create(BaseBox, props);
+        expect(testCase.dom.className).not.toContain('roundedCorners');
+      });
+
+      it('when true adds \'roundedCorners\' class', () => {
```
Comment (jenslonkowski) [inline src/framework/base/box/BaseBox.spec.jsx]: `roundedCorners` can be in double quotation marks.

File: src/framework/base/box/BaseBox.spec.jsx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        const props = {
+          roundedCorners: true,
```
Comment (jenslonkowski) [inline src/framework/base/box/BaseBox.spec.jsx]: The required prop `classes` is missing.

File: src/framework/body/BodyPanel.jsx [hunk 9]
```diff
@@ -1,18 +1,20 @@
...
   });
 
   return (
-    <div className={classes}>
+    <BaseBox classes={classes}>
       {props.children}
-    </div>
+    </BaseBox>
   );
 };
```
Comment (jenslonkowski) [inline src/framework/body/BodyPanel.jsx]: With this implementation it won't be possible to have a BodyPanel with rounded corners as we intended. BodyPanel needs to accept a prop `roundedCorners` and pass it to BasBox.

File: src/framework/box/Box.spec.jsx [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    describe('classes', () => {
+      it('are added to the element', () => {
+        const props = {
+          classes: 'testClass',
+        };
+        const testCase = TestCaseFactory.create(Box, props);
+        expect(testCase.dom.className).toContain(props.classes);
+      });
+
+      it('doesn\'t replace \'box\' class', () => {
```
Comment (jenslonkowski) [inline src/framework/box/Box.spec.jsx]: `box` can be in double quotation marks.

File: src/framework/base/box/BaseBox.jsx (hunk 4 above)
Comment (mohsinhijazee) [inline src/framework/base/box/BaseBox.jsx]: Not sure, but removed. Test passes without it so not needed.

File: src/framework/box/Box.spec.jsx (hunk 10 above)
Comment (mohsinhijazee) [inline src/framework/box/Box.spec.jsx]: Plus, we can use _does not_ instead of _doesn't_ to avoid another single quote everywhere else in the same PR.
